### ISSUE TITLE
[BUG] Sporadic CI failure for ExponentialSmoothing

### ISSUE THREAD

**Issue by hazrulakmal** [sktime/sktime#5026]
**Describe the bug**

I have recently noticed that most of my PR's CI tests have failed due to `ExponentialSmoothing` when running the `TestAllForecasters` test suite, even though my PR has nothing to do with the estimator.

During a discussion in the Discord dev-channel, @fkiraly mentioned that it might be because of a new parameter set that was added at the developer sprint. I am not sure which one it is, so please kindly attach the PR here (it might be PR 4935, but I will let Franz verify it).

**Reference**

some PR CI tests that failed
1. https://github.com/sktime/sktime/actions/runs/5745271056/job/15574945549
2. https://github.com/sktime/sktime/actions/runs/5757122598/job/15608571574?pr=5009
3. https://github.com/sktime/sktime/actions/runs/5684209162/job/15406498704?pr=4877

**Expected behavior**

It should be consistent whether it is meant to pass or fail.

**Comment by fkiraly** [comment]
It looks like it is always parameter set 2, and it sporadically produces `NaN` forecasts.

One suspicion is that it may be due to `use_boxcox=True`.
A vanilla Box-Cox transform will produce `NaN` for negative values.

**Comment by hazrulakmal** [comment]
> A vanilla Box-Cox transform will produce NaN for negative values.

but the y values are all positives by default from `_make_series` as specified in this line https://github.com/sktime/sktime/blob/e58d7e991c7c044536e739d4d3f2314fbdaea54e/sktime/forecasting/tests/test_all_forecasters.py#L687

**Comment by hazrulakmal** [comment]
will it be possible that the reason is from the generation of y series, would specifying a random state = 42 or some int solves the randomness issue?

**Comment by fkiraly** [comment]
maybe - but that wouldn't solve the root cause, which so far looks like a genuine failure?

**Comment by hazrulakmal** [comment]
I have never seen a bug that cannot be reproduced like this. In case anyone wants to give it a try, the failing tests to check are as follows:

```python
import numpy as np
from sktime.forecasting.exp_smoothing import ExponentialSmoothing
from sktime.forecasting.model_selection import temporal_train_test_split

fh_int_oos = np.array([2, 5])
n_columns  = 1 # 2 (sometimes failed for multivariate)
estimator_instance = ExponentialSmoothing(damping_trend=0.28, method='SLSQP', seasonal='add', smoothing_level=0.3, smoothing_trend=0.5, sp=2, trend='add',use_boxcox=True)
y = _make_series(n_columns=n_columns)
y_train, y_test = temporal_train_test_split(y)
estimator_instance.fit(y_train, fh=fh_int_oos)
y_pred = estimator_instance.predict()
```

It failed when y_pred produced a NaN forecast, which only happens sporadically.

My hypothesis is that the issue is related to y_series. There may be some pattern in y_series that causes this, but I can't verify this at the moment since I haven't been able to identify what kind of y_series could cause this behaviour.

**Comment by fkiraly** [comment]
the way I would diagnose is to run a loop over random seeds (to `_make_series`), and terminate at the first failing random seed. That will yield a reproducible example.

**Comment by achieveordie** [comment]
Do you have any updates about this issue? I'd take a look if not (although, I suppose a simple `use_boxcox=False` could fix the symptom, if not the root cause.)

This issue is considered [fixed](https://github.com/statsmodels/statsmodels/issues/6243) in `statsmodel` although there are comments about this being [numerically unstable](https://github.com/statsmodels/statsmodels/issues/6243#issuecomment-554217520). If true, we might have to consider that `_make_series()` might not be at fault here.

**Comment by fkiraly** [comment]
> Do you have any updates about this issue? 

No further insights, sorry. Haven't seen it in a while though.

**Comment by achieveordie** [comment]
In that case, let me take this up and diagnose it for further insights.
